pageserver/src/tenant/timeline/walreceiver/connection_manager.rs

```diff
@@ -606,7 +683,33 @@ impl ConnectionManagerState {
     }
 
     /// Adds another broker timeline into the state, if its more recent than the one already added there for the same key.
-    fn register_timeline_update(&mut self, timeline_update: SafekeeperTimelineInfo) {
+    fn register_timeline_update(&mut self, typed_msg: TypedMessage) {
+        let mut is_discovery = false;
+        let timeline_update = match typed_msg.r#type() {
+            MessageType::SafekeeperTimelineInfo => {
+                let info = typed_msg
+                    .safekeeper_timeline_info
+                    .expect("proto type mismatch from broker message");
```

Comment:
We don't want a broken SK to crash the pageserver. I think it's reasonable to log the error at warn! level and do nothing.